lib/midiHandler.py

```diff
@@ -0,0 +1,295 @@
+__author__ = 'PMarchant'
+
+import os
+import json
+import math
+import pprint
+import logging, logging.handlers, logging.config
+from tracemalloc import BaseFilter
+from music21 import *
+from talkingscores.settings import BASE_DIR, MEDIA_ROOT, STATIC_ROOT, STATIC_URL
+
+logger = logging.getLogger("TSScore")
+
+class MidiHandler:
+    def __init__(self, get, folder, filename):
+        self.queryString = get
+        self.folder = folder
+        self.filename = filename.replace(".mid", "")
+
+    #get list of selected / unselected instruments from binary of number.  Leftmost value is always 1
+    def get_selected_instruments(self):
+        bsi = int(self.queryString.get("bsi"))
+        self.selected_instruments = []
+        while (bsi>1):
+            print ("bsi = " + str(bsi))
```

Comment:
Did you consider using your logger for all the print statements?

Reply:
I like that the logger will output to the log file - which I use for errors etc.  

But I tend to use print to show much more information for debugging as I work - so the log file doesn't get too big.  
I can see them in the console while I am testing - which is useful for scores that take a while to process - I can see how far it is going - while waiting for the error...
I don't actually know where / if pythonanywhere outputs them to?  

Before I commit - I sometimes tidy up some of the print statements and sometimes keep some that I think could still be useful.  
But I gradually miss a few - until there are probably loads that I don't need anymore...  I should go back and tidy them up.  

Or are you saying I could use the logger to output to the console and it would be better than print statements?

Reply:
Fixed for midiHandlery.py in faff2e5 and ccda6c6
logger.debug to output to the console instead of to the log file.
logger format adds line numbers etc too!